### PR TITLE
Fix logic for the Invoice Me radio and option

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Fix logic for the Invoice Me radio and option

Add utility functions for getters
Naming, cleanup and refactoring

[stage-18]

## Motivation and Context
Fix issues with the radio button. Refactoring and clean up.

## How Has This Been Tested?
Tested changes locally. Updated relevant unit test coverage.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc please review. Thanks!

File: web/scripts/billing/services/svc-subscription-factory.js [hunk 1]
```diff
@@ -14,6 +14,42 @@ angular.module('risevision.apps.billing.services')
         factory.apiError = '';
       };
 
+      factory.getItemSubscription = function () {
+        return factory.item && factory.item.subscription || {};
```
Comment (alex-deaconu) [inline web/scripts/billing/services/svc-subscription-factory.js:18]: Name is a bit awkward as to not conflict with the main `getSubscription` function from the API. Thanks.

**Comment by alex-deaconu** [comment]
@santiagonoguez please review. Thanks!

**Review by santiagonoguez — COMMENTED** [review]
Code looks good to me ( even if I never could get to see the invoice radio in my company hahaha ), just a minor not so relevant comment :D

File: web/scripts/billing/services/svc-subscription-factory.js [hunk 2]
```diff
@@ -14,6 +14,42 @@ angular.module('risevision.apps.billing.services')
...
+
+      factory.getPaymentSourceId = function() {
+        if (factory.getItemSubscription().payment_source_id) {
+          return factory.getItemSubscription().payment_source_id;
+        } else if (factory.getItemCustomer().primary_payment_source_id) {
+          return factory.getItemCustomer().primary_payment_source_id;
+        } else {
+          return null;
+        }
+      };
```
[the remark this answers is not in the document]
Comment (alex-deaconu) [reply web/scripts/billing/services/svc-subscription-factory.js:43]: @santiagonoguez yeah you're right, I just copied and kept the same logic as the `auto_collection` above and didn't bother refactoring. Thanks.

**Review by ezequielc — APPROVED** [review]

**Pull request merged**